plugin/testdata/override-ngx-re-test.lua

```diff
@@ -0,0 +1,39 @@
+-- Salt Security Kong Traffic Collector v1.5.14-0.
+-- Copyright (C) 2022 Salt Security Inc.
+local typedefs = require "kong.db.schema.typedefs"
+local re_find = ngx.re.find
+local re_match = ngx.re.match
+
+if (type(re_find) == "nil") or (type(re_match) == "nil")
+    or (type(ngx.re) == "nil") then
+    return {
+        name = "was-nil",
+        fields = {
+            { consumer = typedefs.no_consumer },
+            { protocols = typedefs.protocols_http },
+            { config = {
+                type = "record",
+                fields = {
+                    { uuid = typedefs.uuid },
+                },
+            },
+            },
+        }
+    }
+end
+
+return {
+    name = "override-ngx-re-test",
+    fields = {
+        { consumer = typedefs.no_consumer },
+        { protocols = typedefs.protocols_http },
+        { config = {
+            type = "record",
+            fields = {
+                { uuid = typedefs.uuid },
+            },
+        },
+        },
+    }
+}
+
```

Comment:
This test is very confusing and I feel a little incomplete; let's use a `custom_validator` here to test that local aliases `re_find` and `re_match` work against some fields. You can create fields for the find and the match and pass those to the `custom_validator` function along with another schema field which would be the thing you are searching for.